# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1359793046 -3600
#      Sat Feb 02 09:17:26 2013 +0100
# Node ID c4c46e1a086b44f52d9581ba2aa3db2e96c1acf1
# Parent  9cb64bafa7bd4e06827dae625a9d1c31ca19c36e
gui: switch to console when running an editor file or a command from history

* main-window.cc (handle_command_double_clicked): focus console window

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -161,17 +161,17 @@ main_window::handle_clear_history_reques
 {
   octave_link::post_event (this, &main_window::clear_history_callback);
 }
 
 void
 main_window::handle_command_double_clicked (const QString& command)
 {
   _terminal->sendText (command);
-  _terminal->setFocus ();
+  focus_command_window ();
 }
 
 void
 main_window::open_online_documentation_page ()
 {
   QDesktopServices::openUrl (QUrl ("http://gnu.org/software/octave/doc/interpreter"));
 }
 
